src/Conversion/ONNXToTorch/NN/BinaryOps.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------- SqrtOp.cpp - ONNX Op Transform ------------------===//
+//
+// Copyright 2022, Helprack LLC.
+//
+// =============================================================================
+//
+// This file lowers most binary operators from torch to onnx using a template
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToTorch/NN/CommonUtils.h"
+#include "src/Conversion/ONNXToTorch/ONNXToTorchCommon.hpp"
+
+using namespace mlir;
+using namespace mlir::torch;
+using namespace mlir::torch::Torch;
+
+// Element-wise binary ops lowering to Krnl dialect.
+//===----------------------------------------------------------------------===//
+template <typename ONNXBinaryOp, typename TorchBinaryOp>
+struct ONNXToTorchElementwiseBinaryOpLowering : public ConversionPattern {
```

Comment:
Could this template be used to implement add (#10)?

Reply:
cc @ramakota-helprack

Reply:
Thanks for looking into it. I think #10 is ready then